src/fspec-archive.c

```diff
@@ -61,45 +61,46 @@ main(int argc, char **argv)
     struct archive *a = NULL;
     struct archive_entry *entry = NULL;
     const char *prog;
-    int opt;
 
     prog = argc ? strrchr(argv[0], '/') : NULL;
     prog = prog ? prog + 1 : argv[0];
+    prog = prog ? prog : "";
+
+    argc--;
+    argv++;
 
     a = archive_write_new();
     entry = archive_entry_new();
     if (!a || !entry)
         errx(1, "alloc failure");
 
-    while ((opt = getopt(argc, argv, "x:")) != -1) {
-        switch (opt) {
-        case 'x':
-            if (archive_write_set_format_by_name(a, optarg) != ARCHIVE_OK)
-                errx(1, "%s", archive_error_string(a));
-            break;
-        default:
-            usage(prog);
-        }
-    }
-    argc -= optind;
-    argv += optind;
-
-    if (!archive_format(a) && prog) {
-        if (strcmp(prog, "fspec-tar") == 0)
-            archive_write_set_format_pax_restricted(a);
-        else if (strcmp(prog, "fspec-cpio") == 0)
-            archive_write_set_format_cpio_newc(a);
+    if (strcmp(prog, "fspec-tar") == 0) {
+        archive_write_set_format_pax_restricted(a);
+    } else if (strcmp(prog, "fspec-initcpio") == 0) {
+        archive_write_set_format_cpio_newc(a);
+    } else if (strcmp(prog, "fspec-iso") == 0) {
+        archive_write_set_format_iso9660(a);
+    } else if (strcmp(prog, "fspec-isolinux") == 0) {
+        archive_write_set_format_iso9660(a);
+        if ( archive_write_set_option(a, "iso9660", "boot", "isolinux/isolinux.bin") != ARCHIVE_OK
+          || archive_write_set_option(a, "iso9660", "boot-catalog", "isolinux/boot.cat") != ARCHIVE_OK
```

Comment:
Where do these files come from? Do you have to run `fspec-isolinux` from some particular directory?

Reply:
Nah, they just need to be in your fspec, they come from your syslinux install so its just up to the user. These are just paths embedded into the cd metadata,